### ISSUE TITLE
No Models.model_mia1201.UNet_DA

### ISSUE THREAD

**Issue by lihe-CV** [YonghengSun1997/ODADA#1]
Hello, excuse me,  I did not find the **Models.model_mia1201.UNet_DA** package that includes the model architecture. I would like to take a look at the Orthogonal loss you proposed. I think it is calculated in the model, but I did not find your model architecture code.

**Comment by YonghengSun1997** [comment]
hello, I have uploaded the code to github.&nbsp;Thank you for being interested in our work.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "YonghengSun1997/ODADA"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年12月16日(星期五) 晚上7:44
***@***.***&gt;;
***@***.***&gt;;发送
主题:&nbsp;[YonghengSun1997/ODADA] No Models.model_mia1201.UNet_DA (Issue #1)





 
Hello, excuse me,  I did not find the Models.model_mia1201.UNet_DA package that includes the model architecture. I would like to take a look at the Orthogonal loss you proposed. I think it is calculated in the model, but I did not find your model architecture code.

**Comment by YonghengSun1997** [comment]
These packages are not the essential, you can delete them. If you want to implement the code, please (1) have a dataloader and a code flow from data to output; (2) rewrite your code, especially the model part (another branch for DSR) and the loss function (orthogonal loss).

**Comment by zzs95** [comment]
> > These packages are not the essential, you can delete them. If you want to implement the code, please (1) have a dataloader and a code flow from data to output; (2) rewrite your code, especially the model part (another branch for DSR) and the loss function (orthogonal loss).
> 
> Thanks!

Hi, have you done the fixes for the code? Or successfully reimplemented?

**Comment by YonghengSun1997** [comment]
Thank you for your help in improving this Repository. 
(1) I have uploaded a toy dataset (HK and BIDMC) for reimplement. You can download a full-version dataset via
https://drive.google.com/drive/folders/1KEomtcpTUYCc94nAvEBBsT3vvLnR4rPN?usp=share_link
(2) I have uploaded all the packages needed for reimplement. If any package is missing, please let me know.

**Comment by YonghengSun1997** [comment]
Following SAML [1], we preprocessed the data and only retained the slices with the prostate region.
[1] Liu et al., shape-aware meta-learning for generalizing prostate mri segmentation to unseen domains, MICCAI 2020.

**Comment by zzs95** [comment]
@YonghengSun1997 @hetan697 Great, thank you. But there may be syntax errors in main.py about the `optimizer`, the definition of `dc` and `precision`, it's better to upload again.

**Comment by lllzzzrrr** [comment]
> It looks like we cannot run the code without edit it. How do you deal with the lack of the following package?
> 
> ```python
> from Models.model_mia1201 import UNet_DA
> # import lib
> 
> from utils.dice_loss import get_soft_label, val_dice
> from utils.dice_loss import Intersection_over_Union
> from metrics import LogNLLLoss
> 
> from utils.evaluation import AverageMeter
> from utils.binary import assd, dc, jc, precision, sensitivity, specificity, F1, ACC
> from torch.optim import lr_scheduler
> from data import PrepareDataset, Rescale, ToTensor, Normalize
> 
> import surface_distance as surfdist
> from medpy import metric
> ```
> 
> There are also many **errors** in the `main.py`, like line 159 ![image](https://user-images.githubusercontent.com/50706594/209099917-4e9ca754-cee2-42af-9493-7d66da4b0aa8.png) ![image](https://user-images.githubusercontent.com/50706594/209100016-432126dc-fffa-4a86-8ef6-ed94c02724f7.png)
> 
> I just don't know how to deal with them.

您好！请问您是怎么解决这些引用错误的？比如val_dice，直接删掉的话，后面还用这个计算了b_dice等等。替换的话，在utils/dice_loss.py 里面有val_dice_fetus，val_dice_isic，val_dice_isic_raw三种。像Intersection_over_Union和surface_distance也是，在后面也有用到。

**Comment by wh-isper** [comment]
代码里分割损失使用的是LogNLLLoss()，但论文里说是bce+dice loss，这里也应该改一下把
